copy-as-format.el

```diff
@@ -241,9 +246,16 @@ a language unknown to it.")
           text))
 
 (defun copy-as-format--org-mode (text _multiline)
-  (format "#+BEGIN_SRC %s\n%s\n#+END_SRC\n"
-          (replace-regexp-in-string "-mode\\'" "" (symbol-name major-mode))
-          text))
+  (let* ((file (and (buffer-file-name) (file-name-nondirectory (buffer-file-name)))))
+    (format "%s#+BEGIN_SRC %s %s\n%s\n#+END_SRC\n"
+            (if (and copy-as-format-asciidoc-include-file-name file)
+                (format "#+CAPTION: %s\n" file)
+              "")
+            (replace-regexp-in-string "-mode\\'" "" (symbol-name major-mode))
+            (if copy-as-format-include-line-number
+                (format "-n %s" (line-number-at-pos))
+              "")
+            text)))
```

Comment:
This results in:
```
#+CAPTION: copy-as-format.el
#+BEGIN_SRC emacs-lisp -n 242
(defun copy-as-format--mediawiki (text multiline)
  (format "<syntaxhighlight lang='%s'%s>\n%s\n</syntaxhighlight>"
          (copy-as-format--language)
          (if (not multiline) " inline" "")
          text))
#+END_SRC
```

Which, outputs this in HTML:
<img width="691" alt="Screen Shot 2023-11-13 at 2 49 44 PM" src="https://github.com/sshaw/copy-as-format/assets/17570/12cfbd3e-fd09-4fec-9a34-d8425f23a17d">


Problem is the line number is not correct. That method starts at line 237:
https://github.com/sshaw/copy-as-format/blob/b9f6f725ca9701c5a02bfb479573fdfcce2e1e30/copy-as-format.el#L237

---

Also, if a filename contains underscores, it is not displayed correctly: 

<img width="454" alt="Screen Shot 2023-11-13 at 2 52 20 PM" src="https://github.com/sshaw/copy-as-format/assets/17570/3b9be8c6-35ce-419c-a871-618d5380f536">


Here filename is `ignored_column_tasks.rb`. Not sure how to escape. `\_` did not work :(